src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/ObjectManager.cs

```diff
@@ -773,8 +773,17 @@ internal void CompleteISerializableObject(object obj, SerializationInfo info, St
             {
                 throw new SerializationException(SR.Format(SR.Serialization_ConstructorNotFound, t), e);
             }
+            try
+            {
+                constInfo.Invoke(obj, new object[] { info, context });
+            }
+            // This will only throw TargetInvocationExceptions, but to provide a better exception for dangerous deserialization, unwrap that
+            // and re-wrap it in a DeserializationBlockedException (while preserving its stack)
+            catch (TargetInvocationException outerException) when (outerException.InnerException is DeserializationBlockedException)
+            {
+                throw new DeserializationBlockedException(outerException.InnerException);
```

Comment:
Won't this wrap a `DeserializationBlockedException` within another `DeserializationBlockedException`?

(Aside: I'll open a bug to add an `ExceptionDispatchInfo` property to `TargetInvocationException`. It'll make rethrowing inner exceptions much cleaner.)